src/Components/CookieConsentContext.js

```diff
@@ -3,10 +3,12 @@ import * as React from "react";
 const CookieConsentContext = React.createContext({});
 
 export function CookieConsentProvider(props) {
-  const [cookieConsentChoice, setCookieConsentChoice] = React.useState(() =>
-    getCookieConsent()
+  const [cookieConsentChoice, setCookieConsentChoice] = React.useState(
+    "undecided"
   );
 
+  React.useEffect(() => setCookieConsentChoice(getCookieConsent()), []);
+
```

Comment:
How about to extend `useCookieConsent` that it returns "undecided" `if (typeof window !== "undefined")`? With that `cookieConsentChoice` would be correct from the first render and not only after `React.useEffect`.

Reply:
Given a hydrated DOM should be *exactly* the same SSR and CSR, it is advantageous *not* to initialize the state with "undecided". In other words: since SSR _always_ is `undecided`, the first render _always must be_ undecided, too.